cypress/e2e/05-get_result_from_patient_link.cy.js

```diff
@@ -1,28 +1,85 @@
 const dayjs = require("dayjs");
 const { getPatientLinkByTestEventId } = require("../utils/testing-data-utils");
-const { loginHooks } = require("../support/e2e");
+const {
+  loginHooks,
+  testNumber,
+  generatePatient,
+  generateCovidOnlyDevice,
+} = require("../support/e2e");
 const { frontendURL } = require("../utils/request-utils");
+const {
+  cleanUpPreviousRunSetupData,
+  setupOrgFacility,
+  setupPatient,
+  setupCovidOnlyDevice,
+  cleanUpRunOktaOrgs,
+  setupTestOrder,
+} = require("../utils/setup-utils");
+
+const specRunName = "spec05";
+const currentSpecRunVersionName = `${testNumber()}-cypress-${specRunName}`;
 
 loginHooks();
 describe("Getting a test result from a patient link", () => {
-  let patientLink, patientDOB, patientObfuscatedName;
-  before("retrieve the patient link and dob", () => {
-    cy.task("getTestEventId").then((testEventId) => {
-      getPatientLinkByTestEventId(testEventId).then((res) => {
-        const patientLinkId = res.body.data.testResult.patientLink.internalId;
+  const patient = generatePatient();
+  const covidOnlyDevice = generateCovidOnlyDevice();
+  const patientDOB = patient.dobForPatientLink;
+  const patientObfuscatedName =
+    patient.firstName + " " + patient.lastName[0] + ".";
+  let patientLink;
+
+  before("setup spec data", () => {
+    cy.task("getSpecRunVersionName", specRunName).then(
+      (prevSpecRunVersionName) => {
+        if (prevSpecRunVersionName) {
+          cleanUpPreviousRunSetupData(prevSpecRunVersionName);
+          cleanUpRunOktaOrgs(prevSpecRunVersionName);
+        }
+        let data = {
+          specRunName: specRunName,
+          versionName: currentSpecRunVersionName,
+        };
+        cy.task("setSpecRunVersionName", data);
+        let patientId, createdDeviceId, specimenTypeId, testEventId;
+        setupOrgFacility(currentSpecRunVersionName)
```

Comment:
The setup methods here are all `.then` chained because some of them require variables like `patientId`, `createdDeviceId`, and `specimenTypeId` to be populated from previous setup methods